### PR TITLE
Add ingester flag to customize message in limiter error messages

### LINKED ISSUE #5452: Customize error message when hitting limits
**Is your feature request related to a problem? Please describe.**

Prometheus/Remote write agents that get throttled will get a typical error message like this in the logs

```
server returned HTTP status 400 Bad Request: user=xxx: per-user series limit of 150000 exceeded, please contact administrator to raise it (local limit: 0 global limit: 150000 actual local limit: 30000)
```

**Describe the solution you'd like**
It should be possible to customize error messages for limits or other problems

Ideally we should make the part "please contact administrator to raise it" customizable. So admin can provide more exact info on where to get help.

**Describe alternatives you've considered**
None

**Additional context**
None

### PR DESCRIPTION
**Pull request by kumanik**
**What this PR does**:
Adds an admin contact address flag to the ingester and uses it while printing error messages when hitting limits.

**Which issue(s) this PR fixes**:
Fixes #5452 

**Checklist**
- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by kumanik** [comment]
@friedrichg Have a look 

I was thinking of adding tests for checking contact details at-
https://github.com/kumanik/cortex/blob/81a3cd3c5b5bedb9c7c8208ea7a059fbaf1c7730/pkg/ingester/limiter_test.go#L504C2-L507

Or should I create a completely new test?

File: pkg/ingester/ingester.go [hunk 1]
```diff
@@ -124,6 +124,9 @@ type Config struct {
 
 	// For testing, you can override the address and ID of this ingester.
 	ingesterClientFactory func(addr string, cfg client.Config) (client.HealthAndIngesterClient, error)
+
+	// For admin contact details
+	AdminContactDetails string `yaml:"admin_contact"`
```
Comment (friedrichg) [inline pkg/ingester/ingester.go:129]:
```suggestion
	AdminLimitMessage string `yaml:"admin_limit_message"`
```

File: pkg/ingester/ingester.go [hunk 2]
```diff
@@ -144,6 +147,8 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.Int64Var(&cfg.DefaultLimits.MaxInflightPushRequests, "ingester.instance-limits.max-inflight-push-requests", 0, "Max inflight push requests that this ingester can handle (across all tenants). Additional requests will be rejected. 0 = unlimited.")
 
 	f.StringVar(&cfg.IgnoreSeriesLimitForMetricNames, "ingester.ignore-series-limit-for-metric-names", "", "Comma-separated list of metric names, for which -ingester.max-series-per-metric and -ingester.max-global-series-per-metric limits will be ignored. Does not affect max-series-per-user or max-global-series-per-metric limits.")
+
+	f.StringVar(&cfg.AdminContactDetails, "ingester.admin-contact", "", "Add site administrator contact details for customising error messages")
```
Comment (friedrichg) [inline pkg/ingester/ingester.go:151]:
```suggestion
	f.StringVar(&cfg.AdminLimitMessage, "ingester.admin-limit-message", defaultAdminLimitMessage, "Customize the message contained in limit errors")
```
Where defaultAdminLimitMessage is a string var with "please contact administrator to raise it" that can be used in tests

File: pkg/ingester/limiter.go [hunk 3]
```diff
@@ -121,36 +124,52 @@ func (l *Limiter) formatMaxSeriesPerUserError(userID string) error {
 	actualLimit := l.maxSeriesPerUser(userID)
 	localLimit := l.limits.MaxLocalSeriesPerUser(userID)
 	globalLimit := l.limits.MaxGlobalSeriesPerUser(userID)
+	adminContact := l.adminContact
+	if len(l.adminContact) > 0 {
+		adminContact = "(" + l.adminContact + ") "
+	}
```
Comment (friedrichg) [inline pkg/ingester/limiter.go:130]:
```suggestion
```
This is not needed

File: pkg/ingester/limiter.go [hunk 4]
```diff
@@ -121,36 +124,52 @@ func (l *Limiter) formatMaxSeriesPerUserError(userID string) error {
...
 
-	return fmt.Errorf("per-user series limit of %d exceeded, please contact administrator to raise it (local limit: %d global limit: %d actual local limit: %d)",
-		minNonZero(localLimit, globalLimit), localLimit, globalLimit, actualLimit)
+	return fmt.Errorf("per-user series limit of %d exceeded, please contact administrator %sto raise it (local limit: %d global limit: %d actual local limit: %d)",
+		minNonZero(localLimit, globalLimit), adminContact, localLimit, globalLimit, actualLimit)
```
Comment (friedrichg) [inline pkg/ingester/limiter.go:133]:
```suggestion
	return fmt.Errorf("per-user series limit of %d exceeded, %s (local limit: %d global limit: %d actual local limit: %d)",
		minNonZero(localLimit, globalLimit), adminLimitMessage, localLimit, globalLimit, actualLimit)
```

**Review by friedrichg — CHANGES_REQUESTED** [review]
Thank you for your contribution!

You don't need to add any additional tests, just keep the current tests working.

File: docs/configuration/config-file-reference.md [hunk 5]
```diff
@@ -2620,6 +2620,10 @@ instance_limits:
 # max-global-series-per-metric limits.
 # CLI flag: -ingester.ignore-series-limit-for-metric-names
 [ignore_series_limit_for_metric_names: <string> | default = ""]
+
+# Flag to add site administrator contact details for customising error messages
+# CLI flag: -ingester.admin-contact
```
Comment (friedrichg) [inline docs/configuration/config-file-reference.md:2625]: needs to change

**Review by friedrichg — COMMENTED** [review]
Good job @kumanik , the only thing missing is a FEATURE in the CHANGELOG.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,6 +1,7 @@
 # Changelog
 
 ## master / unreleased
+* [FEATURE] Ingester: added `-admin-limit-message` to customize the message contained in limit errors.#5452
```
Comment (friedrich-at-adobe) [inline CHANGELOG.md:4]:
```suggestion
* [FEATURE] Ingester: added `-admin-limit-message` to customize the message contained in limit errors.#5460
```

**Comment by kumanik** [comment]
It is done I suppose

**Review by friedrichg — APPROVED** [review]
Thank you!

**Comment by kumanik** [comment]
No, thank you. Really glad I could contribute.

**Pull request merged**